ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/prepare-and-run-synthea.py

```diff
@@ -165,19 +170,25 @@ def run_synthea(synthea_folder_filepath, benes_to_generate, future_months):
     """
     
     logfile_path = f'{synthea_folder_filepath}synthea-' + time.strftime("%Y_%m_%d-%I_%M_%S_%p") + '.log'
+    synthea_failed = False
     if future_months > 0:
         print(f'Running synthea ({synthea_folder_filepath}national_bfd_v2.sh) with {benes_to_generate} benes and {future_months} future months...')
-        output = subprocess.check_output(shlex.split(f'{synthea_folder_filepath}national_bfd_v2.sh {benes_to_generate} {future_months}'), text=True, stderr=subprocess.STDOUT)
+        process_cmd = shlex.split(f'{synthea_folder_filepath}national_bfd_v2.sh {benes_to_generate} {future_months}')
     else:
         print(f'Running synthea ({synthea_folder_filepath}national_bfd.sh) with {benes_to_generate} benes...')
-        output = subprocess.check_output(shlex.split(f'{synthea_folder_filepath}national_bfd.sh {benes_to_generate}'), text=True, stderr=subprocess.STDOUT)
-    with open(logfile_path, 'w') as f:
-        f.write(output)
+        process_cmd = shlex.split(f'{synthea_folder_filepath}national_bfd.sh {benes_to_generate}')
+    with subprocess.Popen(process_cmd, text=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, bufsize=1) as p, \
+        open(logfile_path, 'w') as f:
+        for line in p.stdout:
+            print(line, end='')
```

Comment:
Yeah, this isn't super obvious -- you suggested that instead of printing the full Synthea generation log (which mostly consisted of messages indicting a module was loaded) we should log a "ping" or some other message to indicate that work is still being done. Fortunately, the `national_bfd*.sh` scripts both print when they're generating beneficiaries (and thus executing the Synthea `.jar` to the log, which I figured would work well as an indicator of work.

To clarify, I've modified the `prepare-and-run-synthea.py` script log to `STDOUT` _and_ the log file. This will _always_ include logs from the Python script itself and the `national_bfd*.sh` script that was ran, but will _conditionally_ include the logs logged from running the Synthea `.jar` if the verbose flag (`-v`) you mentioned is *enabled*. Relevant lines are lines 12 - 16 in `run_synthea`:

```bash
# Only log to stdout (and therefore a file) if verbosity was requested. This would be the -v
# flag in docker_entrypoint.sh
if [ -z ${VERBOSE_SYNTHEA_LOGS+x} ]; then
  exec >/dev/null 2>&1
fi
```